controllers/packetmachine_controller.go

```diff
@@ -375,23 +371,6 @@ func (r *PacketMachineReconciler) reconcile(ctx context.Context, machineScope *s
 		result = ctrl.Result{RequeueAfter: 10 * time.Second}
 	case infrav1.PacketResourceStatusRunning:
 		log.Info("Machine instance is active", "instance-id", machineScope.GetInstanceID())
-
-		// TODO: see if this can be removed with kube-vip in place
-		// This logic is here because an elastic ip can be assigned only an
-		// active node. It needs to be a control plane and the IP should not be
-		// assigned to anything at this point.
-		controlPlaneEndpoint, _ = r.PacketClient.GetIPByClusterIdentifier(
-			machineScope.Cluster.Namespace,
-			machineScope.Cluster.Name,
-			machineScope.PacketCluster.Spec.ProjectID)
-		if len(controlPlaneEndpoint.Assignments) == 0 && machineScope.IsControlPlane() {
-			if _, _, err := r.PacketClient.DeviceIPs.Assign(dev.ID, &packngo.AddressStruct{
-				Address: controlPlaneEndpoint.Address,
-			}); err != nil {
-				log.Error(err, "err assigining elastic ip to control plane. retrying...")
-				return ctrl.Result{RequeueAfter: time.Second * 20}, nil
-			}
-		}
```

Comment:
@cprivitere Should I make this configurable with an env var? For example:

```go
if os.env("EIP_MANAGEMENT" == "CPEM" {
    controlPlaneEndpoint, _ = r.PacketClient.GetIPByClusterIdentifier(
			machineScope.Cluster.Namespace,
			machineScope.Cluster.Name,
			machineScope.PacketCluster.Spec.ProjectID)
		if len(controlPlaneEndpoint.Assignments) == 0 && machineScope.IsControlPlane() {
			if _, _, err := r.PacketClient.DeviceIPs.Assign(dev.ID, &packngo.AddressStruct{
				Address: controlPlaneEndpoint.Address,
			}); err != nil {
				log.Error(err, "err assigining elastic ip to control plane. retrying...")
				return ctrl.Result{RequeueAfter: time.Second * 20}, nil
			}
		}
	}
```

Reply:
Same here, this seems a good place to start, if nothing else. Let's start with it and after I do some deployments I can comment if there's more tweaking to do.